File: py/bubble_sort.py
from typing import List


def bubble_sort(arr: List[int]):
    for i in range(0, len(arr) - 1):
        for j in range(i + 1, len(arr)):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]


if __name__ == "__main__":
    arr = [30, 50, 20, 10, 40, 90, 80, 60, 70]
    bubble_sort(arr)
    print(arr)

File: go/bubble_sort.go
package main

import "fmt"

func RunBubbleSort() {
	arr := []int{64, 25, 12, 22, 11}
	bubbleSort(arr)
	fmt.Println(arr)
}

func bubbleSort(arr []int) {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		for j := i + 1; j < n; j++ {
			if arr[i] > arr[j] {
				arr[j], arr[i] = arr[i], arr[j]
			}
		}
		fmt.Printf("Iteration(%d) arr: %v\n", i+1, arr)
	}
}
